smartsim/entity/model.py

```diff
@@ -328,13 +329,23 @@ def colocate_db_tcp(
             "debug": debug,
             "db_identifier": db_identifier,
         }
+        # test
         self._set_colocated_db_settings(tcp_options, common_options, **kwargs)
 
     def _set_colocated_db_settings(
         self,
-        connection_options: t.Dict[str, t.Any],
-        common_options: t.Dict[str, t.Any],
-        **kwargs: t.Any,
+        connection_options: t.Mapping[str, t.Union[int, list[str], str]],
```

Comment:
ditto "pedantic, would rather update `t.List[T]` -> `list[T]` all at once in a future ticket" comment here